src/main/java/gift/global/config/RestTemplateConfig.java

```diff
@@ -0,0 +1,41 @@
+package gift.global.config;
+
+import gift.global.exception.RestTemplateException;
+import java.time.Duration;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.web.client.RestTemplate;
+
+@Configuration
+public class RestTemplateConfig {
+
+    @Bean
+    public RestTemplate restTemplate(RestTemplateBuilder builder) {
+        return builder
+            .setConnectTimeout(Duration.ofSeconds(5)) // 연결
+            .setReadTimeout(Duration.ofSeconds(5)) // 읽기
+            .additionalInterceptors(clientHttpRequestInterceptor())
+            .build();
+    }
+
+    /**
+     *  RestTemplate 이 HTTP 요청 보낼때마다 요청 가로챔
+     */
+    public ClientHttpRequestInterceptor clientHttpRequestInterceptor() {
+        return ((request, body, execution) -> {
+            RetryTemplate retryTemplate = new RetryTemplate();
+            retryTemplate.setRetryPolicy(new SimpleRetryPolicy(3)); // 최대 3번 재시도
+            try {
+                return retryTemplate.execute(context -> execution.execute(request, body)); // HTTP 요청 실행
+            } catch (Throwable throwable) {
+                throw new RestTemplateException();
```

Comment:
이렇게 하면 예외 메시지가 끊길 것 같은데요, `RestTemplateException`에 부모 메시지도 얹어주면 좋을 것 같아요! RestTemplateException 클래스도 변경이 필요하겠군요!
```suggestion
                throw new RestTemplateException(throwable);
```

Reply:
부모 메시지도 얹도록 수정하였습니다!
![image](https://github.com/user-attachments/assets/6da7a827-b272-44f4-b0ca-e71a6e103af1)
![image](https://github.com/user-attachments/assets/df801140-40b7-4467-a654-1dff7c08bde1)